File: mtc-model/src/main/scala/org/drugis/mtc/PriorStartingValueGenerator.scala
package org.drugis.mtc

class PriorStartingValueGenerator[M <: Measurement, P <: Parametrization[M]](
model: NetworkModel[M, P])
extends StartingValueGenerator[M] {
	override def getBaselineEffect(study: Study[M]) = 0.0
	override def getRandomEffect(study: Study[M], p: BasicParameter) = 0.0
	override def getRelativeEffect(p: BasicParameter) = 0.0
	override def getRandomEffectsVariance() = 0.25
}

File: mtc-data/src/main/java/org/drugis/mtc/parameterization/PriorStartingValueGenerator.java
package org.drugis.mtc.parameterization;

import org.drugis.mtc.model.Study;
import org.drugis.mtc.model.Treatment;

public class PriorStartingValueGenerator implements StartingValueGenerator {
	final PriorGenerator d_priorGen;
	
	public PriorStartingValueGenerator(PriorGenerator priorGen) {
		d_priorGen = priorGen;
	}
	
	public double getRelativeEffect(Study study, BasicParameter p) {
		return 0.0;
	}

	public double getRelativeEffect(BasicParameter p) {
		return 0.0;
	}

	public double getTreatmentEffect(Study study, Treatment treatment) {
		return 0.0;
	}

	public double getStandardDeviation() {
		return d_priorGen.getRandomEffectsSigma() / 2.0;
	}
}